src/btree/bt_handle.c

```diff
@@ -437,6 +437,22 @@ __btree_conf(WT_SESSION_IMPL *session, WT_CKPT *ckpt)
 		FLD_SET(btree->assert_flags, WT_ASSERT_COMMIT_TS_KEYS);
 	else if (WT_STRING_MATCH("never", cval.str, cval.len))
 		FLD_SET(btree->assert_flags, WT_ASSERT_COMMIT_TS_NEVER);
+
+	/*
+	 * A durable timestamp always implies a commit timestamp. But never
+	 * having a durable timestamp does not imply anything about a commit
+	 * timestamp.
+	 */
+	WT_RET(__wt_config_gets(session,
+	    cfg, "assert.durable_timestamp", &cval));
+	if (WT_STRING_MATCH("always", cval.str, cval.len))
+		FLD_SET(btree->assert_flags,
+		    WT_ASSERT_COMMIT_TS_ALWAYS | WT_ASSERT_DURABLE_TS_ALWAYS);
+	else if (WT_STRING_MATCH("key_consistent", cval.str, cval.len))
+		FLD_SET(btree->assert_flags, WT_ASSERT_DURABLE_TS_KEYS);
+	else if (WT_STRING_MATCH("never", cval.str, cval.len))
+		FLD_SET(btree->assert_flags, WT_ASSERT_DURABLE_TS_NEVER);
```

Comment:
While reviewing the test, it occurs to me that we don't check for non-sensical combinations with the `commit_timestamp` setting. Something like `durable_timestamp=always,commit_timestamp=never` should get an error (I think that might be the only one actually).

Reply:
The error check for the combination `durable_timestamp=always, commit_timestamp=never` is actually part of the `commit_transaction` API.  `commit_timestamp` is optional for non-prepared transactions only, and if `durable_timestamp` is specified for non-prepared transaction, error will be returned. So, we thought no need to check this condition as part of assert.  If you feel strongly that it is better we add this check as part of assert, will do it.

Reply:
Thanks for that explanation. That's fine.